### PR TITLE
feat: Make sleep cancellable

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
**Summary**  
Replaces the old fire-and-forget `sleep` with a **cancellable** delay: the returned promise includes **`cancel()`**, which clears the timer and either **rejects** with **`PromiseCancellationError`** (configurable), **resolves** when `cancelError: null`, or uses **string / `Error`** overrides like `withTimeout`.

**API**  
- **`sleep(ms)`** or **`sleep({ ms, cancelError? })`** — runtime checks: finite **`number`**, or **plain object** with finite **`ms`** (includes `Object.create(null)`).  
- **`CancellablePromise<T>`** — generic **`Promise<T> & { cancel(): void }`** (default `T` is `void`; **`sleep`** returns **`CancellablePromise<void>`**).  
- **`SleepArg`**, **`SleepOptions`**, **`PromiseCancellationError `** exported from the package surface.

**Tests**  
Coverage for timing, cancel/reject/resolve paths, **`cancelError`** variants, invalid arguments, and plain-object edge cases.

### CODE REVIEW

File: lib/asyncbox.ts [hunk 1]
```diff
@@ -11,12 +18,66 @@ export class TimeoutError extends Error {
   }
 }
 
+/** Thrown when a promise is cancelled via `cancel`. */
+export class PromiseCancellation extends Error {
```
Comment (eglitise) [inline lib/asyncbox.ts]: Since this is an error, I think the name should be adjusted to `PromiseCancellationError`

File: lib/types.ts [hunk 2]
```diff
@@ -36,6 +36,35 @@ export interface LongSleepOptions {
...
+ * Object form of {@link sleep}'s argument: duration plus optional cancellation rejection override.
+ */
+export interface SleepOptions {
+  /** Duration in milliseconds */
+  ms: number;
+  /**
+   * When {@link sleep}'s `cancel` runs: a non-empty string becomes {@link PromiseCancellation}
+   * with that message; an `Error` rejects with that same instance; `null` resolves the promise
+   * instead of rejecting; omitted or other falsy values (except `null`) use the default
+   * {@link PromiseCancellation}.
```
[the remark this answers is not in the document]
Comment (mykola-mokhnach) [reply lib/types.ts]: updated

File: lib/types.ts [hunk 3]
```diff
@@ -36,6 +36,35 @@ export interface LongSleepOptions {
...
+   */
+  cancelError?: string | Error | null;
+}
+
+/**
+ * Argument to {@link sleep}: either milliseconds or a plain options object (see {@link SleepOptions}).
+ */
+export type SleepArg = number | SleepOptions;
+
```
[the remark this answers is not in the document]
Comment (mykola-mokhnach) [reply lib/types.ts:58]: fixed

File: lib/asyncbox.ts (hunk 1 above)
Comment (mykola-mokhnach) [reply lib/asyncbox.ts]: updated

File: lib/asyncbox.ts [hunk 4]
```diff
@@ -11,12 +18,67 @@ export class TimeoutError extends Error {
   }
 }
 
+/** Thrown when a promise is cancelled via `cancel`. */
+export class PromiseCancellationError extends Error {
+  constructor(message: string = 'Promise cancelled') {
+    super(message);
+    this.name = 'PromiseCancellationError';
+  }
+}
```
[the remark this answers is not in the document]
Comment (mykola-mokhnach) [reply lib/asyncbox.ts:27]: updated

File: lib/asyncbox.ts [hunk 5]
```diff
@@ -11,12 +18,67 @@ export class TimeoutError extends Error {
...
+    timeoutId = setTimeout(resolve, ms);
+  }) as CancellablePromise<void>;
+
+  promise.cancel = () => {
+    if (timeoutId) {
+      clearTimeout(timeoutId);
+      timeoutId = undefined;
+    }
+    if (cancelError === null) {
+      resolveFn?.();
```
[the remark this answers is not in the document]
Comment (mykola-mokhnach) [reply lib/asyncbox.ts]: updated

**Review by KazuCocoa — APPROVED** [review]
